ledger/metrics/src/main/scala/com/daml/telemetry/Tracing.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.telemetry
+
+import io.opentelemetry.api.GlobalOpenTelemetry
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.TextMapPropagator
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object Tracing {
+
+  def encodeTraceMetadata(context: Context): Map[String, String] = {
+    val buffer = mutable.Map[String, String]()
+    GlobalOpenTelemetry.getPropagators.getTextMapPropagator
+      .inject(context, buffer, TracingMetadataSetter)
+
+    buffer.toMap
+  }
+
+  def decodeTraceMetadata(data: java.util.Map[String, String]): Option[Context] = {
+    if (data == null) {
```

Comment:
why would this ever be null?

Reply:
No idea, TBH, I more or less copy-pasted it from the other repo, where it's been developed over the months. First of all, I would change it to `Option` if needed, which I'm going to do in my next PR with the improvements to the API.